mech/__main__.py

```diff
@@ -242,11 +332,14 @@ def main(args=None):
 
     elif arguments['ssh']:
         name = arguments.get("--user")
-        if not name:
```

Comment:
You'll likely need to put this back once the username becomes optional in the mechfile.